sklearn/cluster/_agglomerative.py

```diff
@@ -930,6 +945,16 @@ def _fit(self, X):
         """
         memory = check_memory(self.memory)
 
+        self._metric = self.metric
+        # TODO(1.4): Remove
+        if self.affinity != "deprecated":
+            warnings.warn(
+                "Attribute `affinity` was deprecated in version 1.2 and will be removed"
+                " in 1.4. Use `metric` instead",
+                DeprecationWarning,
```

Comment:
The reason we use `FutureWarning` is that is appears by default when the function is imported from another file.

```suggestion
                FutureWarning,
```